src/pages/[redirect]/index.tsx

```diff
@@ -1,85 +1,50 @@
 import Head from 'next/head';
-import Link from 'next/link';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
 
-import { TINY_SITE } from '@/constants/url';
+import LoaderTimer from '@/components/Redirect/LoaderTimer';
+import NotFound from '@/components/Redirect/NotFound';
+import RedirectFooter from '@/components/Redirect/RedirectFooter';
 import fetchOriginalUrl from '@/utils/fetchOriginalUrl';
 
-import RedirectIcon from '../../components/icons/redirect';
-
-interface LoaderTimerProps {
-    timer: number;
-    goButtonClickHandler: () => void;
-}
-
-const LoaderTimer = ({ timer, goButtonClickHandler }: LoaderTimerProps) => {
-    if (timer < 1) {
-        return (
-            <div className="mt-4 flex flex-col items-center space-y-2">
-                <RedirectIcon />
-                <p className="text-1xl">Redirecting...</p>
-            </div>
-        );
-    } else {
-        return (
-            <>
-                <div className="loader border-t-4 rounded-full border-gray-500 animate-spin aspect-square w-20 flex justify-center items-center text-yellow-700 text-4xl font-bold mt-4">
-                    {timer}
-                </div>
-                <button onClick={goButtonClickHandler} className="mt-4 p-2 bg-blue-500 text-white rounded-md">
-                    Go
-                </button>
-            </>
-        );
-    }
-};
-
-const RedirectFooter = () => (
-    <div className="absolute bottom-0 right-0 p-2 text-gray-500 w-screen flex justify-center items-center">
-        <Link
-            className="text-sm text-gray-400 font-bold cursor-pointer hover:underline"
-            href={TINY_SITE}
-            target="_blank"
-            rel="noopener noreferrer"
-        >
-            By <span className="font-bold">Real Dev Squad</span>
-        </Link>
-    </div>
-);
-
 const Redirect = () => {
     const router = useRouter();
     const { redirect: shortUrlCode } = router.query as { redirect: string };
     const [originalUrl, setOriginalUrl] = useState('');
     const [timer, setTimer] = useState(5);
     const [isPremiumUser] = useState(false);
     const [showTooltip, setShowTooltip] = useState(false);
+    const [notFound, setNotFound] = useState(false);
```

Comment:
Can you decrease the number of usestate

Reply:
Can you suggest to me how to decrease it?

Reply:
removed isPremiumUser useState

Reply:
either consolidates the component's state into a single state object, or currently I am looking into state management library which one to choose I have Zotai, Zustand, SWR, Redux, Recoil in mind

Reply:
This conversion should be resolved, and we can discuss which library to use on a call.